tests/tests_geomstats/test_matrices.py

```diff
@@ -1,328 +1,573 @@
-        skew_mat = gs.array([[0, -2.0], [2.0, 0]])
-        result = self.space.is_skew_symmetric(skew_mat)
-        expected = gs.array(True)
-        self.assertAllClose(result, expected)
-
-        not_a_sym_mat = gs.array([[1.0, 0.6, -3.0], [6.0, -7.0, 0.0], [0.0, 7.0, 8.0]])
-        result = self.space.is_skew_symmetric(not_a_sym_mat)
-        expected = gs.array(False)
-        self.assertAllClose(result, expected)
-
-    @geomstats.tests.np_autograd_and_tf_only
-    def test_is_symmetric_vectorization(self):
-        points = gs.array(
-            [
-                [[1.0, 2.0], [2.0, 1.0]],
-                [[3.0, 4.0], [4.0, 5.0]],
+from geomstats.geometry.matrices import Matrices, MatricesMetric
+from tests.conftest import Parametrizer, TestCase, TestData
+
+SQRT_2 = math.sqrt(2)
+
+EYE_2 = [[1.0, 0], [0.0, 1.0]]
+MINUS_EYE_2 = [[-1.0, 0], [0.0, -1.0]]
+EYE_3 = [[1.0, 0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]
+MAT1_23 = [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]
+MAT2_23 = [[0.0, -2.0, -3.0], [0.0, 1.0, 1.0]]
+MAT1_33 = [[1.0, 2.0, 3.0], [2.0, 4.0, 5.0], [3.0, 5.0, 6.0]]
+MAT2_33 = [[1.0, 2.0, 3.0], [2.0, 4.0, 7.0], [3.0, 5.0, 6.0]]
+MAT3_33 = [[0.0, 1.0, -2.0], [-1.0, 0.0, -3.0], [2.0, 3.0, 0.0]]
+MAT4_33 = [[2.0, -1.0, 0.0], [-1.0, 2.0, -1.0], [0.0, -1.0, 2.0]]
+MAT5_33 = [[2.0, 0.0, 0.0], [1.0, 3.0, 0.0], [8.0, -1.0, 2.0]]
+MAT6_33 = [[1.0, 3.0, 4.0], [0.0, 2.0, 6.0], [0.0, 0.0, 2.0]]
+MAT7_33 = [[0.0, 0.0, 0.0], [1.0, 0.0, 0.0], [8.0, -1.0, 0.0]]
+MAT8_33 = [[0.0, 3.0, 4.0], [0.0, 0.0, 6.0], [0.0, 0.0, 0.0]]
+
+
+class TestMatrices(TestCase, metaclass=Parametrizer):
+    class TestDataMatrices(TestData):
+        def belongs_data(self):
+            sq_mat = EYE_2
+            smoke_data = [
+                dict(m=2, n=2, mat=sq_mat, expected=True),
+                dict(m=2, n=1, mat=sq_mat, expected=False),
+                dict(m=2, n=3, mat=[MAT1_23, MAT2_23], expected=[True, True]),
+                dict(m=2, n=1, mat=MAT1_23, expected=False),
+                dict(
+                    m=3,
+                    n=3,
+                    mat=[MAT1_33, MAT2_33, MAT3_33],
+                    expected=[True, True, True],
+                ),
+            ]
+            return self.generate_tests(smoke_data)
+
+        def equal_data(self):
+
+            smoke_data = [
+                dict(m=2, n=2, mat_1=EYE_2, mat_2=EYE_2, expected=True),
+                # dict(m=2, n=2, mat_1=EYE_2, mat_2=EYE_3, expected=False),
+                dict(m=2, n=3, mat_1=MAT1_23, mat_2=MAT2_23, expected=False),
+            ]
+            return self.generate_tests(smoke_data)
+
+        def mul_data(self):
+            mats_1 = (
                 [[1.0, 2.0], [3.0, 4.0]],
+                [[-1.0, 2.0], [-3.0, 4.0]],
+                [[1.0, -2.0], [3.0, -4.0]],
+            )
+            mats_2 = [[[2.0], [4.0]], [[1.0], [3.0]], [[1.0], [3.0]]]
+            mat_1_X_mat_2 = [[[10.0], [22.0]], [[5.0], [9.0]], [[-5.0], [-9.0]]]
```

Comment:
no uppercase letter in variables' names as per PEP8: use `mat_1_x_mat_2` or `mat_1_mul_mat_2`